pkg/syncer/spec/mutators/deployment.go

```diff
@@ -64,9 +70,50 @@ func (dm *DeploymentMutator) Mutate(downstreamObj *unstructured.Unstructured) er
 	// Alias for the template.spec to improve readability.
 	templateSpec := &deployment.Spec.Template.Spec
 
+	desiredServiceAccountName := "default"
+	if templateSpec.ServiceAccountName != "" && templateSpec.ServiceAccountName != "default" {
+		desiredServiceAccountName = templateSpec.ServiceAccountName
+	}
+
+	secretList, err := dm.secretLister.List(labels.Everything())
```

Comment:
dm.secretLister comes from the upstream (kcp) informer, so we need to exclude secrets for other logical clusters. We're still working on scoped listers, so in the meantime, I'd recommend adding an indexer on secrets to index by cluster name, then list from the secret indexer (not lister) using the custom index.

Reply:
can you take a look at the implementation? thanks!